### DISCUSSION TITLE
Why get native frames through PerfEvents::walk when cstack=vm && event_type == PERF_SAMPLE ?

### DISCUSSION THREAD

**Discussion by byte0100** [async-profiler/async-profiler#994] [Q&A]
When cstack=vm, StackWalker::walkVM() would do NATIVE_FRAME stack trace. 
So, why we need to call PerfEvents::walk() with cstack=vm  in Profiler::getNativeTrace() ?

**Comment by apangin** [comment]
Thank you for your interest in async-profiler.
If you experience issues with existing async-profiler behavior or suppose a bug in the code, please submit a new [report](https://github.com/async-profiler/async-profiler/issues/new?assignees=&labels=bug&projects=&template=bug-report.yml&title=%28short+issue+description%29), describing expected vs. actual behavior, detailed reproduction steps and any additional information that may assist in troubleshooting.
If you have a proposal how to improve the code, feel free to submit a Pull Request.

**Reply by byte0100** [reply]
Thanks for your reply. I would submit a new report or  a Pull Request.
